drivers/usb/udc/udc_mcux_ehci.c

```diff
@@ -34,19 +36,30 @@ LOG_MODULE_REGISTER(udc_mcux, CONFIG_UDC_DRIVER_LOG_LEVEL);
 
 #define PRV_DATA_HANDLE(_handle) CONTAINER_OF(_handle, struct udc_mcux_data, mcux_device)
 
+/* Required by DEVICE_MMIO_NAMED_* macros */
+#define DEV_CFG(_dev) ((const struct udc_mcux_config *)(_dev)->config)
+#define DEV_DATA(_dev) ((struct udc_mcux_data *)(udc_get_private(_dev)))
+
 struct udc_mcux_config {
 	const usb_device_controller_interface_struct_t *mcux_if;
 	void (*irq_enable_func)(const struct device *dev);
 	void (*irq_disable_func)(const struct device *dev);
 	size_t num_of_eps;
 	struct udc_ep_config *ep_cfg_in;
 	struct udc_ep_config *ep_cfg_out;
-	uintptr_t base;
+	DEVICE_MMIO_NAMED_ROM(reg_base);
 	const struct pinctrl_dev_config *pincfg;
 	usb_phy_config_struct_t *phy_config;
+	const struct device *usb_clock_dev;
+	clock_control_subsys_t usb_clock_subsys;
+	clock_control_subsys_rate_t usb_clock_rate;
+	const struct device *usbphy_clock_dev;
+	clock_control_subsys_t usbphy_clock_subsys;
+	clock_control_subsys_rate_t usbphy_clock_rate;
```

Comment:
For other NXP mcu platforms, it seems that the clock is initialized in the `soc.c`, could you put the clock related codes there?